src/core_write.cpp

```diff
@@ -88,7 +88,12 @@ string ScriptToAsmStr(const CScript& script, const bool fAttemptSighashDecode)
         }
         if (0 <= opcode && opcode <= OP_PUSHDATA4) {
             if (vch.size() <= static_cast<vector<unsigned char>::size_type>(4)) {
-                str += strprintf("%d", CScriptNum(vch, false).getint());
+                if ( (lastopcode == OP_NAME_NEW || lastopcode == OP_NAME_UPDATE || lastopcode == OP_NAME_FIRSTUPDATE )
+                    && vch.size() > static_cast<vector<unsigned char>::size_type>(0) ) {
```

Comment:
`vch` is a `std::vector<unsigned char>`, for which `size() > 0` and `!empty()` is equivalent.  I would prefer the code to use the latter, as it is more expressive and easier to read for this case.